### PR TITLE
Fuel Based Silicon Crucible

### PR DESCRIPTION
**Pull request by MEEPofFaith**
_I'm gonna go ahead and make my suggestion myself._

https://github.com/Anuken/Mindustry-Suggestions/issues/2460

Stats

https://user-images.githubusercontent.com/54301439/119780178-579d1080-be7e-11eb-9e1e-8d368188d108.mp4

(If fuel isn't required, the crafter won't even accept fuel input)

TODO:
- [X] Add stats for fuel item, fuel use, and fuel capacity
  - [X] Make stat/description wording better(?)
  - [X] Find way to convey that the Attributes thing is in fact subtracting fuel use, have "-" without being red.
- [X] Optimize my code. It's a mess.
  - [X] O p t I m I z e   e v e n   m o r e .
- [ ] Current stats are placeholders, maybe find better stats? I suck at balance so I'll leave it to the comments to handle this.

### CODE REVIEW

**Comment by MEEPofFaith** [comment]
TODO:
- [ ] Add stats for fuel item, fuel use, and fuel capacity
- [ ] Current stats are placeholders, maybe find better stats? I suck at balance so I'll leave the comments on this handle that.

**Comment by Quezler** [comment]
[brings back memories](https://github.com/Anuken/Mindustry/pull/927#issuecomment-547309421) 👀

**Comment by MEEPofFaith** [comment]
> [brings back memories](https://github.com/Anuken/Mindustry/pull/927#issuecomment-547309421) 👀
> 
> i like your implementation though 😇

huh didn't know that existed

**Comment by Quezler** [comment]
well yours is a lot more user experience friendly 😊

**Comment by genNAowl** [comment]
Maybe "Not Required" Instead of "Fuel Not Needed"?  "Fuel Not Needed" is a bit wordy.

**Comment by MEEPofFaith** [comment]
Make "me" a smelter...?

**Comment by Anuken** [comment]
To be honest, I'm just not interested in changing mechanics of blocks in the base game that already work fine. This implementation is quite long, likely contains bugs and requires several new bundle entries. I'd rather work on other, new content instead.

**Comment by Iloveidea** [comment]
No storage pyratite? Interesting.

**Comment by Anuken** [comment]
Closing for reason posted earlier.

**Comment by MEEPofFaith** [comment]
i'll probably include something using this code in PM

**Comment by MEEPofFaith** [comment]
And use it I sure did!

https://user-images.githubusercontent.com/54301439/130191477-62b4e2c9-0a6f-41fd-880b-a862174e425d.mp4

**Comment by genNAowl** [comment]
Fat smelter flame
